packages/webapp/pages/settings/organization/[orgId]/members.tsx

```diff
@@ -133,13 +144,50 @@ export const OrganizationMembers = ({
 };
 
 const Page = (): ReactElement => {
-  const { push, query } = useRouter();
+  const router = useRouter();
+  const { query, push } = router;
   const { displayToast } = useToastNotification();
+  const { showPrompt } = usePrompt();
   const { user } = useAuthContext();
   const { organization, role, isFetching } = useOrganization(
     query.orgId as string,
   );
 
+  const { mutateAsync: leaveOrganization, isPending: isLeavingOrganization } =
+    useMutation({
+      mutationFn: async () => {
+        await gqlClient.request(LEAVE_ORGANIZATION_MUTATION, {
+          id: organization.id,
+        });
+      },
+      onSuccess: async () => {
+        router?.replace(webappUrl);
```

Comment:
Maybe we can also remove the query from cache using `queryClient.removeQueries`?